### PR TITLE
Add a lint plugin to ensure we only use ES5 syntax.

### PR DESCRIPTION
**Pull request by samelhusseini**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Fixes https://github.com/google/blockly/issues/3153

### Proposed Changes

Adds an eslint plugin that ensure we don't use ES6 syntax in our codebase.
Rename ``Blockly.utils.object.values`` to ``Blockly.utils.object.valuesShim``.
Use ``Blockly.utils.object.mixin`` for the last instance of Object.assign in our codebase.

### Reason for Changes

No Object.assign in IE. Avoid having to constantly deal with this in the future.

### Test Coverage
Playground.
eslint.

Tested on:

### CODE REVIEW

**Comment by RoboErikG** [comment]
Let's hold off on submitting until we finish discussing options.

**Comment by rachel-fenichel** [comment]
@RoboErikG I'm fine with this (with the plugin set as warn rather than error)

**Comment by RoboErikG** [comment]
Leaving as warnings sgtm.

**Comment by samelhusseini** [comment]
Updated, leaving ``es5/no-es6-methods`` as warn.

File: tests/node/run_node_test.js [hunk 1]
```diff
@@ -22,28 +22,28 @@
 var assert = require('chai').assert;
 var Blockly = require('../../dist/');
 
-var xmlText = `<xml xmlns="https://developers.google.com/blockly/xml">
```
Comment (rachel-fenichel) [inline tests/node/run_node_test.js]: You could also add this file to the eslintignore and leave it as es6.

**Pull request merged**